CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/SpellChecker.java

```diff
@@ -0,0 +1,305 @@
+                        } else {
+                            popupContent.getChildren().addAll(suggestions, labelMessage);
+                        }
+                    }
+
+                    popup.getContent().add(popupContent);
+                    popup.setAutoFix(true);
+                    popup.show(textArea, event.getScreenX(), event.getScreenY() + 10);
+                }
+            } catch (SecurityException | InvocationTargetException | NoSuchMethodException | IllegalAccessException ex) {
+                logAndDisplayErrorMessage("Error while populating suggestions. Spell checking may not be functioning properly.", ex);
+            }
+        }
+    }
+
+    /**
+     * Retrieve the word/phrase under the cursor and check if it is misspelled.
+     * If it is misspelled the index is populated.
+     */
+    private boolean isWordUnderCursorMisspelled() throws InvocationTargetException, NoSuchMethodException, SecurityException, IllegalAccessException {
+        final int cursorIndex = textArea.getCaretPosition();
+
+        if (cursorIndex <= 0 || cursorIndex >= textArea.getText().length()) {
+            //= is to avoid the scenario of displaying the suggesttions of the first/last word
+            // (if they are incorrect) when clicking on the empty space right/below the text
+            return false;
+        }
+
+        for (final Object match : matches) {
+            if (match.getClass() == RuleMatch) {
+
+                final int start = (int) RuleMatch.getMethod("getFromPos").invoke(match);
+                final int end = (int) RuleMatch.getMethod("getToPos").invoke(match);
+                if (cursorIndex >= start && cursorIndex <= end) {
+                    indexOfMisspelledTextUnderCursor = misspells.indexOf(textArea.getText().substring(start, end));
+                    startOfMisspelledTextUnderCursor = start;
+                    endOfMisspelledTextUnderCursor = end;
+                    specificRuleId = (String) RuleMatch.getMethod("getSpecificRuleId").invoke(match);
+                    labelMessage.setText((String) RuleMatch.getMethod("getMessage").invoke(match));
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Prevents highlighting while still typing. When a highlighted word is
+     * corrected manually it'll be marked as correct, similar to that in
+     * Microsoft Word.
+     * @param newText
+     * @return
+     */
+    public boolean canCheckSpelling(final String newText) {
+        final int caretPosition = textArea.getCaretPosition();
+        if (caretPosition == 0) {
+            return true;
+        } else if (caretPosition <= newText.length()) {
+            final String charAtCaret = Character.toString(textArea.getText().charAt(caretPosition - 1));
+            return !newText.isEmpty() && (textArea.isWordUnderCursorHighlighted(caretPosition - 1) || !charAtCaret.matches("[a-zA-Z0-9']"));
```

Comment:
for the regex here, can this pattern be pre-compiled? i.e. use a Pattern object for the regex